src/jobs/helm_push.yaml

```diff
@@ -0,0 +1,137 @@
+  helm_dir:
+    type: string
+    description: "directory containing the helm chart"
+    default: ".helm"
+  app_version:
+    type: string
+    description: "app version in Chart.yaml"
+    default: ""
+  api_version:
+    type: string
+    description: "api version in Chart.yaml"
+    default: ""
+  chart_version:
+    type: string
+    description: "chart version in Chart.yaml"
+    default: ""
+  image_tag:
+    type: string
+    description: "image tag in values.yaml"
+    default: ""
+  charts_repo_domain_ssm:
+    type: string
+    description: "charts repo domain ssm parameter"
+    default: "/codacy/chartmuseum/DOMAIN"
+  charts_repo_user_ssm:
+    type: string
+    description: "charts repo username ssm parameter"
+    default: "/codacy/chartmuseum/BASIC_AUTH_USER"
+  charts_repo_pass_ssm:
+    type: string
+    description: "charts repo password ssm parameter"
+    default: "/codacy/chartmuseum/BASIC_AUTH_PASS"
+  aws_profile:
+    type: string
+    description: "AWS profile to be used"
+    default: "default"
+  region:
+    type: string
+    description: "aws region for the deployment"
+    default: eu-west-1
+  credentials_file_content:
+    type: string
+    default: |
+      [default]
+      aws_access_key_id=$ACCESS_KEY_ID
+      aws_secret_access_key=$SECRET_ACCESS_KEY
+      [integration]
+      source_profile = default
+      role_arn = arn:aws:iam::$AWS_ACCOUNT_ID_INTEGRATION:role/$CONTINUOUS_DELIVERY_ROLE
+      [staging]
+      source_profile = default
+      role_arn = arn:aws:iam::$AWS_ACCOUNT_ID_STAGING:role/$CONTINUOUS_DELIVERY_ROLE
+      [production]
+      source_profile = default
+      role_arn = arn:aws:iam::$AWS_ACCOUNT_ID_PRODUCTION:role/$CONTINUOUS_DELIVERY_ROLE
+
+environment:
+  CHART_NAME: << parameters.chart_name >>
+  HELM_DIR: << parameters.helm_dir >>
+  APP_VERSION: << parameters.app_version >>
+  API_VERSION: << parameters.api_version >>
+  CHART_VERSION: << parameters.chart_version >>
+  IMAGE_TAG: << parameters.image_tag >>
+  AWS_PROFILE: << parameters.aws_profile >>
+  AWS_DEFAULT_REGION: << parameters.region >>
+  CHARTS_REPO_USER_SSM: << parameters.charts_repo_user_ssm >>
+  CHARTS_REPO_PASS_SSM: << parameters.charts_repo_pass_ssm >>
+  CHARTS_REPO_SSM: << parameters.charts_repo_domain_ssm >>
+
+steps:
+  - attach_workspace:
+      at: ~/
+  - run:
+      name: Setup AWS Credentials
+      command: |
+        mkdir -p ~/.aws && touch ~/.aws/credentials
+        cat >> ~/.aws/credentials \<< EOF
+        << parameters.credentials_file_content >>
+        EOF
+  - run:
+      name: Set chart and versions, and set image tag
+      command: |
+        function set_value() {
+          local KEY=$1
+          local VALUE=$2
+          local FILE=$3
+          sed -rie "s|(^\s+?${KEY}\s+?:\s+?)(\S+)|\1${VALUE}|g" ${FILE}
+        }
+
+        CHART_FILE="${HELM_DIR}/${CHART_NAME}/Chart.yaml"
+        VALUES_FILE="${HELM_DIR}/${CHART_NAME}/values.yaml"
+
+        if [ -z $APP_VERSION ] ; then
```

Comment:
instead of binding parameters:
from user -> to env vars -> to bash scripts

please use orbs conditionals and avoid unuseful re-bind
e.g.:
https://github.com/codacy/codacy-orbs/blob/master/src/jobs/checkout_and_version.yml#L23

Reply:
It is a good idea. I'll look into it.